tensorflow/compiler/tf2xla/kernels/image_resize_ops.cc

```diff
@@ -503,7 +504,12 @@ void GeneralCompile(XlaOpKernelContext* ctx, bool align_corners_,
   }
 
   // Output is always type float if 'is_kernel_bilinear' is true.
-  if (is_kernel_bilinear) {
+  // GPU with integer input also uses float, because XLA
+  // integer convolution on CuDNN is either not supported or not allowed
+  // directly.
+  xla::PrimitiveType original_input_type = input_type;
+  if (is_kernel_bilinear || (is_gpu_device &&
+                             xla::primitive_util::IsIntegralType(input_type))) {
```

Comment:
I do not think there is any reason to check for gpu_device, I am pretty sure the whole algorithm only works on floats in the first place. The CPU backend tests are failing anyway.

Reply:
The original code forces F32 only for bilinear, but not for nearest neighbor.  It makes sense to me that nearest neighbor's kernels are made of 0 and 1, therefore float is not necessary.
This change forces F32 for nearest neighbor for GPU in addition to bilinear for all targets.